app/importers/activity_insight_importer.rb

```diff
@@ -225,7 +225,20 @@ def pub_attrs(pub)
     end
 
     def ai_users
-      @users ||= Nokogiri::XML(ai_users_xml).css('Users User').map { |u| ActivityInsightListUser.new(u) }
+      @users = nil
+      users_xml = Nokogiri::XML(ai_users_xml)
+      @users ||= users_xml.css('Users User').map { |u| ActivityInsightListUser.new(u) }
```

Comment:
There is an issue with the use of `||=` here, which is that on the first line of this method, `@users` is assigned to nil, so that means the calls on lines 229 and 230 will never be memoized.

I would probably change this to the following...
```ruby
def ai_users
  return @users if @users.present?

  users_xml = Nokogiri::XML(ai_users_xml)
  @users = users_xml.css('Users User').map { |u| ActivityInsightListUser.new(u) }
```

then check my comment below for how to handle the return value at the end of the rescue block

Reply:
yes, I like the guard clause, will update. thanks 👍🏻